src/renderer/components/command-palette/command-container.tsx

```diff
@@ -0,0 +1,87 @@
+
+import "./command-container.scss";
+import { action, observable } from "mobx";
+import { observer } from "mobx-react";
+import React from "react";
+import { Dialog } from "../dialog";
+import { EventEmitter } from "../../../common/event-emitter";
+import { subscribeToBroadcast } from "../../../common/ipc";
+import { CommandDialog } from "./command-dialog";
+import { CommandRegistration, commandRegistry } from "../../../extensions/registries/command-registry";
+import { clusterStore } from "../../../common/cluster-store";
+import { workspaceStore } from "../../../common/workspace-store";
+import { Cluster } from "../../../main/cluster";
+
+export type CommandDialogEvent = {
+  component: React.ReactElement
+};
+
+const commandDialogBus = new EventEmitter<[CommandDialogEvent]>();
```

Comment:
I'm okay with this for now, but we should reconsider the way how to change children within `CommandOverlay`. It feels a bit strange that we have `commandDialogBus` and some event emitters for this purpose. 

@ixrock Do you see how we can pass children to this component without inner `CommandDialogEvent` manipulations?

Reply:
"Public" api for this is `CommandOverlay. open(<component)`. Bus is only used internally.